python/revng/internal/daemon/graphql.py

```diff
@@ -243,14 +243,22 @@ async def resolve_run_analyses_list(_, info, *, name: str, options: str | None =
         parse_options = json.loads(options) if options is not None else {}
         result = await run_in_executor(manager.run_analyses_list, name, parse_options)
 
-        if result:
-            real_result = result.unwrap()
-            new_index = await run_in_executor(manager.get_context_commit_index)
-            await invalidation_queue.send(Invalidation(new_index, str(real_result.invalidations)))
-            return Diff(json.dumps(real_result.result))
-        else:
+        if not result:
             return result.error.unwrap()
 
+        real_result = result.unwrap()
+        new_index = await run_in_executor(manager.get_context_commit_index)
+        await invalidation_queue.send(Invalidation(new_index, str(real_result.invalidations)))
+
+        # NOTE: kind of a hack, the daemon should not know about what analyses
+        #       revng provides
+        # Since the first initial auto analysis populates the bulk of the
+        # model, it's a good time to save to avoid losing the progress made
+        if name == "renvg-initial-auto-analysis" and current_index < 5:
+            await run_in_executor(manager.save)
```

Comment:
This quite an ugly hack. Can't the client ask to save?

Reply:
There isn't an API for that. I can add it if needed.
However it could be used abusively IMHO.

Reply:
I'd add an API, this is too much of an hack.

Reply:
Ok, adding

Reply:
Implemented